tests/integration/App.test.js

```diff
@@ -1,26 +1,29 @@
 /* global describe, it, expect, jest, jasmine */
-import React from 'react';
 
 import { LegacyWallet, SegwitP2SHWallet } from '../../class';
 
-global.net = require('net');
 global.crypto = require('crypto'); // shall be used by tests under nodejs CLI, but not in RN environment
 const assert = require('assert');
 
-jest.mock('react-native-qrcode-svg', () => 'Video');
-jest.useFakeTimers();
+global.net = require('net'); // needed by Electrum client. For RN it is proviced in shim.js
+const BlueElectrum = require('../../BlueElectrum'); // so it connects ASAP
 
-jest.mock('amplitude-js', () => ({
-  getInstance() {
-    return {
-      init: jest.fn(),
-      logEvent: jest.fn(),
-    };
-  },
-}));
+afterAll(async () => {
+  // after all tests we close socket so the test suite can actually terminate
+  BlueElectrum.forceDisconnect();
+  return new Promise(resolve => setTimeout(resolve, 10000)); // simple sleep to wait for all timeouts termination
+});
+
+beforeAll(async () => {
+  // awaiting for Electrum to be connected. For RN Electrum would naturally connect
+  // while app starts up, but for tests we need to wait for it
+  await BlueElectrum.waitTillConnected();
+});
 
 describe('unit - LegacyWallet', function() {
   it('serialize and unserialize work correctly', () => {
+    jest.setTimeout(30000);
```

Comment:
If it makes sense you can move it to `beforeEach` and should work fine

Reply:
done